tests/crypto/test_cases/test_aead.c

```diff
@@ -0,0 +1,575 @@
+extern test_vector_aead_t __stop_test_vector_aead_chachapoly_data[];
+extern test_vector_aead_t __start_test_vector_aead_chachapoly_simple_data[];
+extern test_vector_aead_t __stop_test_vector_aead_chachapoly_simple_data[];
+
+#define NUM_BUFFER_OVERFLOW_TEST_BYTES (2)                                                      /**< Number of bytes to be used in the overflow test for AEAD. */
+#define AEAD_MAC_SIZE                  (16)                                                     /**< Size of the AEAD MAC. */
+#define AEAD_MAX_TESTED_NONCE_SIZE     (128)                                                    /**< Input buffer size for the AEAD noce. */
+#define AEAD_PLAINTEXT_BUF_SIZE        (265)                                                    /**< Input buffer size for the AEAD plaintext. */
+#define AEAD_PLAINTEXT_BUF_SIZE_PLUS   AEAD_PLAINTEXT_BUF_SIZE + NUM_BUFFER_OVERFLOW_TEST_BYTES /**< Input buffer size for the AEAD plaintext, including 2 buffer overflow bytes. */
+#define AEAD_MAX_MAC_SIZE_PLUS         AEAD_MAC_SIZE + NUM_BUFFER_OVERFLOW_TEST_BYTES           /**< Input buffer size for the AEAD MAC, including 2 buffer overflow bytes. */
+
+#define AEAD_KEY_SIZE_BITS      (256)
+#define AEAD_KEY_SIZE           (AEAD_KEY_SIZE_BITS / 8)
+
+static uint8_t m_aead_input_buf[AEAD_PLAINTEXT_BUF_SIZE];                                       /**< Buffer for holding the AEAD input plaintext or ciphertext. */
+static uint8_t m_aead_output_buf[AEAD_PLAINTEXT_BUF_SIZE_PLUS];                                 /**< Buffer for holding the AEAD output plaintext or ciphertext. */
+static uint8_t m_aead_expected_output_buf[AEAD_PLAINTEXT_BUF_SIZE];                             /**< Buffer for holding the AEAD expected plaintext or ciphertext. */
+static uint8_t m_aead_output_mac_buf[AEAD_MAX_MAC_SIZE_PLUS];                                   /**< Buffer for holding the AEAD generated MAC. */
+static uint8_t m_aead_expected_mac_buf[AEAD_MAX_MAC_SIZE_PLUS];                                 /**< Buffer for holding the AEAD expected MAC. */
+static uint8_t m_aead_key_buf[AEAD_KEY_SIZE];                                                   /**< Buffer for holding the AEAD key. */
+static uint8_t m_aead_ad_buf[AEAD_PLAINTEXT_BUF_SIZE];                                          /**< Buffer for holding the AEAD associated data. */
+static uint8_t m_aead_nonce_buf[AEAD_MAX_TESTED_NONCE_SIZE];                                    /**< Buffer for holding the AEAD nonce data. */
+
+static test_vector_aead_t *p_test_vector;
+/* Some tests require overriding the test vector crypt direction.
+ * Use this additional flag to force decrypt (if set to false).
+ */
+static bool g_encrypt = true;
+static size_t key_len;
+static size_t key_bits;
+static size_t ad_len;
+static size_t mac_len;
+static size_t nonce_len;
+static size_t input_len;
+static size_t output_len;
+
+__attribute__((noinline)) static void clear_buffers(void)
```

Comment:
Why the noinline?

Reply:
My guess: To ensure that it is callable by dwarf symbol from host side

Reply:
Then we should instead make it non-static and give it a unique name, aead_clear_buffers.

Reply:
@frkv is right.

It is named the same in each `test_{cipher}.c` allowing shared test logic on the host side.

Reply:
It's not clear to me how the host is using this, but I'll take it on good faith that this makes sense if it is understood by you.

Reply:
I think that's a good point.
Changed this to use a prefix per file as suggested in b06d84e3